edb/ir/utils.py

```diff
@@ -498,6 +499,49 @@ def flt(n: irast.Call) -> bool:
     return bool(ast.find_children(ir, irast.Call, flt, terminate_early=True))
 
 
+def is_singleton_set_of_call(
+    call: irast.Call
+) -> bool:
+    # Some set functions and operators are allowed in singleton mode
+    # as long as their inputs are singletons
+
+    return call.func_shortname in {
+        sn.QualName('std', 'IN'),
+        sn.QualName('std', 'NOT IN'),
+        sn.QualName('std', 'EXISTS'),
+        sn.QualName('std', '??'),
+        sn.QualName('std', 'IF'),
+    }
```

Comment:
We *could* make this a new flag on operators that we configure in the standard library creation code. No sure if it is worth it, though?